server/src/main/java/org/opensearch/tasks/TaskResourceStats.java

```diff
@@ -31,31 +33,36 @@
  * @opensearch.internal
  */
 public class TaskResourceStats implements Writeable, ToXContentFragment {
-    private final Map<String, TaskResourceUsage> resourceUsage;
+    private final Map<String, AbstractTaskResourceUsage> resourceUsage;
```

Comment:
Please keep `private final Map<String, TaskResourceUsage> resourceUsage;` and handle thread related stats separably. Otherwise, the mixed clusters, it is going to blow (since older clusters would not know how to deal with thread stats).

Reply:
@reta The `TaskResourceStats` object is what represents the `resource_stats` field in the Tasks APi response, but currently `TaskResourceStats` only handles `TaskResourceUsage` objects which is why I am making the change here to a parent class so we can handle both `TaskResourceUsage` and `TaskThreadUsage` together. 

Are you suggesting I make `TaskThreadUsage` extend `TaskResourceUsage` instead?
I did briefly explore moving thread stats completely out of `resource_stats` as well but that also has backwards compatibility challenges and more importantly functionally it makes more sense to me to keep it in `resource_stats`.

Reply:
> Are you suggesting I make TaskThreadUsage extend TaskResourceUsage instead? Or modify the constructor of TaskResourceStats to handle thread stats separately?

Exactly, we need to handle `TaskThreadUsage` separately during serialization / deserialization (see please comments below), otherwise mixed clusters would fail to process such responses.

Reply:
Got it. Will work on the change. Thank you!